src/chip/isa.rs

```diff
@@ -0,0 +1,304 @@
+//! The supported Instruction Set Architecture for a chip specification
+
+use serde::{de::Deserializer, Deserialize, Serializer};
+
+use std::collections::HashMap;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+pub struct Isa {
+    /// The degree-0 (i.e. single qubit) hardware objects and their
+    /// supported instructions.
+    #[serde(rename(deserialize = "1Q"))]
+    #[serde(rename(serialize = "1Q"))]
+    pub one_q: HashMap<String, Option<OneQ>>,
+    /// The degree-1 (i.e. two qubit) hardware objects and their
+    /// supported instructions.
+    #[serde(rename(deserialize = "2Q"))]
+    #[serde(rename(serialize = "2Q"))]
+    pub two_q: HashMap<String, Option<TwoQ>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub metadata: Option<Metadata>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+pub struct Metadata {}
```

Comment:
Should this include a flattened map of miscellaneous fields, so that it's round-trip-able?

https://serde.rs/attr-flatten.html#capture-additional-fields